templates/designate/config/designate.conf

```diff
@@ -20,11 +20,12 @@ healthcheck_enabled=True
 connection = mysql+pymysql://${DBUSER}:${DBPASSWORD}@${DBHOST}/${DB}
 
 [storage:sqlalchemy]
-connection = mysql+pymysql://root:${DBPASSWORD}@${DBHOST}/${DB}?charset=utf8
```

Comment:
OK so this one is quite strange, and the use of "root" for the username seems intentional here.

So we would guess there's some anomalous behavior when running migrations (still using sqlalchemy-migrate??) or later on where somehow the regular DB user doesnt have enough privs.   which would be very unusual for mysql/mariadb,  this is a straightforward kind of DB for which we give full grants to the user and if there's something that doesnt work with the regular user, we should fix that (and I least need to know what it is).

But if this *does* need to be root then we cant use `${DBPASSWORD}` here, we would need to add a new token `${DBROOTPASSWORD}` and then pass that in from osp-secret under the `DatabasePassword` selector, but even then this is kind of not really planned in the overall architecture and we really need this to be a non-root user.

Reply:
im 99% sure this was just some remnant of something from ancient history, also the utf8 part we dont need since we create the database with utf8 encoding by default in any case, the clients we use these days are generally utf8 oriented overall ever since Python 3 made Unicode mandatory.